static/tests/frontend/specs/commentIconsGroups.js

```diff
@@ -0,0 +1,232 @@
+          if ($commentLine.children().length !== 1) {
+            return false;
+          }
+        }
+        return true;
+      }).done(done);
+    });
+
+    context("when user clicks the SM icon again", function () {
+      before(function () {
+        helper.padInner$("sm_icon").click();
+      });
+
+      it("groups the icons", function (done) {
+        var $commentLine = getCommentLineOfId(commentIds[SECOND_ICON_GROUP]);
+        firstIconIsCorrect($commentLine, function () {
+          otherIconsAreCorrect($commentLine, done);
+        });
+      });
+    });
+  });
+
+  context("when user adds lines", function(){
+    before(function(done){
+      var $lastIcon = getIconOfId(commentIds[LAST_ICON]);
+      var $previousIconCommentLine = $lastIcon.parent();
+      createTwoLinesOnTopOfScript(function(){
+        var lastLine = SCENE_HEADING_LINE + 2;
+        waitForIconToBeMoved($lastIcon, $previousIconCommentLine, done);
+      });
+    });
+
+    after(function (){
+      utils.undo();
+    });
+
+    it("first icon group should remain grouped", function(done){
+      var $commentLine = getCommentLineOfId(commentIds[FIRST_ICON_GROUP]);
+      commentLineIsGroupedCorrectly($commentLine, FIRST_GROUP_LENGTH, done);
+    });
+
+    it("second icon group should remain grouped", function (done){
+      var $commentLine = getCommentLineOfId(commentIds[SECOND_ICON_GROUP]);
+      commentLineIsGroupedCorrectly($commentLine, SECOND_GROUP_LENGTH, done);
+    })
+  });
+
+  context("when comment is deleted in first group", function(){
+    before(function(done){
+      apiUtils.simulateCallToDeleteComment(commentIds[FIRST_ICON_GROUP+1]);
+      helper.waitFor(function(){
+        return getCommentLineOfId(commentIds[FIRST_ICON_GROUP]).children().length === 1;
+      }).done(done);
+    });
+
+    after(function(){
+      utils.undo();
+    });
+
+    it("should ungroup icons", function(done){
+      var $icon = getIconOfId(commentIds[FIRST_ICON_GROUP])
+      helper.waitFor(function(){
+        return !$icon.hasClass("with-ellipsis") && $icon.css("opacity") === "1";
+      }).done(done);
+    });
+
+  });
+
+  context("when comment is deleted in second group", function(){
+    before(function(done){
+      apiUtils.simulateCallToDeleteComment(commentIds[SECOND_ICON_GROUP]);
+      helper.waitFor(function(){
+        console.log(getCommentLineOfId(commentIds[SECOND_ICON_GROUP + 1]).children().length)
+        return getCommentLineOfId(commentIds[SECOND_ICON_GROUP + 1]).children().length === 2;
+      }).done(done);
+    });
+
+    after(function(){
+      utils.undo();
+    });
+
+    it("should remain grouped", function(done){
+      var $commentLine = getCommentLineOfId(commentIds[SECOND_ICON_GROUP+1]) // first icon of second group was deleted
+      commentLineIsGroupedCorrectly($commentLine, SECOND_GROUP_LENGTH - 1, done);
+    });
+
+  });
+});
+
+// ----
+
+function getIconOfId(commentId){
```

Comment:
Senti um pouco de dificuldade com alguns nomes de função. Por exemplo:

```suggestion
function getCommentIcon(commentId){
```